util/coverage/functest_coverage.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from pathlib import Path
+from pprint import pformat
+from typing import List
+
+from device_profile_data import extract_profile_data
+from common import (
+    LLD_TARGET,
+    BazelTestType,
+    CoverageParams,
+    run,
+)
+
+# Commands that this script uses
+LLVM_PROFDATA = "llvm-profdata"
+
+
+def handle_libs(device_libs_all: List[str]) -> List[str]:
+    """Filter device libraries that are not compatible with the target.
+
+    Args:
+        device_libs_all: A list of device libraries.
+
+    Returns:
+        `device_libs_all` with incompatible libraries filtered out.
+    """
+    # Remove on_host libs
+    device_libs_incompat = [lib for lib in device_libs_all if "on_host" in lib]
+    logging.info(f"incompatible libraries: {pformat(device_libs_incompat)}")
+    # TODO: may want to add the coverage runtime to avoid undefined symbol warnings
+    return sorted(list(set(device_libs_all) - set(device_libs_incompat)))
+
+
+def handle_objs(merged_library: Path, obj_files: List[str]) -> None:
+    """Create a library from the given object files.
+
+    Args:
+        merged_library: Path where to save the merged library.
+        obj_files: A list of object files.
+    """
+    run(LLD_TARGET, "--warn-unresolved-symbols", "-zmuldefs", "-o",
```

Comment:
Nit: I might add a comment explaining why we need to allow multiple definitions :)

Reply:
Thanks @dmcardle, done.